x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+  private soClient?: SavedObjectsClient;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '1m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              const runAt = new Date().toISOString();
+
+              if (this.soClient && this.esClient) {
+                const finder = this.soClient.createPointInTimeFinder<StoredSLO>({
+                  type: SO_SLO_TYPE,
+                  perPage: 1000,
+                  fields: ['id'],
+                });
+                let searchAfterKey: SortResults | undefined;
+                const soIds: string[] = [];
+                let sloDefIdsToDelete: string[] = [];
+
+                for await (const response of finder.find()) {
+                  const soItems = response.saved_objects.map((so) => so.attributes.id);
+
+                  const { sloDefIds, searchAfter } = await this.fetchSloDefinitions(searchAfterKey);
+                  searchAfterKey = searchAfter;
+
+                  const { missingSloDefIds, missingSOIds } = findMissingItems(
+                    soItems.concat(soIds),
+                    sloDefIds.concat(sloDefIdsToDelete)
+                  );
+
+                  soIds.push(...missingSOIds);
+                  sloDefIdsToDelete = missingSloDefIds;
+                }
+                this.logger.info(
+                  JSON.stringify({
+                    idsToDelete: Array.from(new Set(sloDefIdsToDelete)),
+                    soIds: Array.from(new Set(soIds)),
+                  })
+                );
+
+                if (sloDefIdsToDelete.length > 0) {
+                  await this.esClient.deleteByQuery({
+                    index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+                    query: {
+                      terms: {
+                        'slo.id': sloDefIdsToDelete,
+                      },
+                    },
+                  });
+                }
+              }
+
+              return { state: { lastRunAt: runAt } };
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  fetchSloDefinitions = async (searchAfter?: SortResults) => {
+    if (this.esClient) {
+      const result = await this.esClient.search({
+        size: 1000,
+        index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+        fields: ['slo.id'],
+        collapse: {
+          field: 'slo.id',
+        },
+        _source: false,
+        sort: [
+          {
+            'slo.id': {
+              order: 'desc',
+            },
+          },
+        ],
```

Comment:
you probably want to sort by slo.id, slo.revision, slo.instanceId` since one slo.id could have many slo.revision, and many slo.instanceId (if partitioned by), and I guess we want the search request to sort consistently across these fields